SwiftLeeds/Views/Components/FancyHeaderView.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  HeaderView.swift
+//  SwiftLeeds
+//
+//  Created by Kannan Prasad on 05/07/2022.
+//
+
+import SwiftUI
+
+struct FancyHeaderView: View {
+    
+    internal init(title: String, foregroundImageURL: URL?, backgroundImageURL: URL?) {
+        self.title = title
+        self.foregroundImageURL = foregroundImageURL
+        self.backgroundImageURL = backgroundImageURL
+        self.forgroundImageName = nil
+        self.backgroundImageName = nil
+    }
+    
+    internal init(title: String, forgroundImageName: String?, backgroundImageName: String?) {
+        self.title = title
+        self.foregroundImageURL = nil
+        self.backgroundImageURL = nil
+        self.forgroundImageName = forgroundImageName
+        self.backgroundImageName = backgroundImageName
+    }
+    
+    let title: String
+    let foregroundImageURL: URL?
+    let backgroundImageURL: URL?
+    let forgroundImageName: String?
+    let backgroundImageName: String?
+    
+    private let foregroundImageWidth = 160.0
+    private let aspectRatio = 1.66
+    var shadowColor: Color {
+       Color.black.opacity(1/3)
+    }
+    
+    @State var foregroudGroupViewHeight: CGFloat = .zero
+    
+    var body: some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .edgesIgnoringSafeArea(.top)
+            .aspectRatio(aspectRatio, contentMode: .fill)
+            .background(backgroundImage
+                            .aspectRatio(contentMode: .fill))
+            .overlay(foregroudGroup,alignment: .center)
+            .padding(.bottom,foregroudGroupViewHeight/2)
+    }
+    
+    private var backgroundImage: some View {
+        if let backgroundImageURL = backgroundImageURL {
+           return AnyView(AsyncImage(url: backgroundImageURL) { phase in
+                switch phase {
+                case .empty:
+                    ProgressView()
+                case .success(let image):
+                    crateRectangleImage(for: image, aspectRatio: aspectRatio)
+                case .failure(_):
+                    AnyView(crateRectangleImage(for: Image(Assets.Image.playhouseImage),
```

Comment:
Thanks for this wonderful review comment. I've been writing Siwft UI for my side projects and learning, but it never went for a peer review, so I may have been following the Xcode default suggestions for many of the problems which have other side effects.

Reply:
Happy to help! It's easy to miss these things when you don't ship it in production.

I often find that with these sorts of things it can be better to look around online, rather than just throwing in the xcode suggestion - you never know when you'll end up with something like this!